test/test_sympy_utils.py

```diff
@@ -520,6 +523,74 @@ def test_z3_proof_floordiv_eq_simplify(self):
         r = solver.check()
         self.assertEqual(r, z3.unsat)
 
+class TestSingletonInt(TestCase):
+    def test_basic(self):
+        j1 = SingletonInt(1, 1)
```

Comment:
The two arg constructor here seems pretty error prone. Can we make them kwarg only?

Reply:
Updated to be kwarg only.